02-04-domain-validations/internal/platform/server/handler/courses/search_all.go

```diff
@@ -0,0 +1,38 @@
+package courses
+
+import (
+	"net/http"
+
+	mooc "github.com/CodelyTV/go-hexagonal_http_api-course/02-04-domain-validations/internal"
+	"github.com/gin-gonic/gin"
+)
+
+type CourseResponse struct {
```

Comment:
If you're going to use this struct only within this package, then there's no need to declare it as a public type.